### ISSUE TITLE
Detect unused `# fmt: off` & `# fmt: on` comments

### ISSUE THREAD

**Issue by jack-mcivor** [astral-sh/ruff#17098]
### Summary

From the format documentation https://docs.astral.sh/ruff/formatter/#format-suppression

> As such, adding # fmt: on and # fmt: off comments within expressions will have no effect. In the following example, both list entries will be formatted, despite the # fmt: off:
> ```python
> [
>     # fmt: off
>     '1',
>     # fmt: on
>     '2',
> ]
> ```

Should there be a new rule to detect such cases? I think it's in the spirit of an unused noqa comment, for which there is RUF100 https://docs.astral.sh/ruff/rules/unused-noqa/.

**Comment by MichaReiser** [comment]
That's an interesting idea. Not quiet sure how I'd go about implementing it

**Comment by jack-mcivor** [comment]
Ah this is implemented already in RUF028 https://docs.astral.sh/ruff/rules/invalid-formatter-suppression-comment/ https://github.com/astral-sh/ruff/pull/9899

**Comment by jack-mcivor** [comment]
Yeah got you. I'm migrating a large codebase from black->ruff. Actually both would be useful for me so that I could do two steps:

- Remove unused suppression comments (many are invalid but the code is formatted fine)
- For all invalid suppression comments, decide what to do (either lift into the nearest statement, or remove the suppression)

So not a burning desire to have this, but feel free to reopen
